File: 0024-swap-nodes-in-pairs/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy=new ListNode(0);
        dummy.next=head;
        ListNode curr=dummy;
        ListNode temp;
        while(curr.next!=null && curr.next.next!=null )
        {
            temp=curr.next.next;
            curr.next.next = temp.next;
            temp.next = curr.next;
            curr.next=temp;
            curr=curr.next.next;;
        }
        return dummy.next;
    }
}

File: 0024-swap-nodes-in-pairs/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapPairs(struct ListNode* head) {
    if(head==NULL || head->next==NULL)
        return head;
    struct ListNode* dummy=(struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next=head->next;
    struct ListNode* temp=NULL;
    struct ListNode* prev=NULL;
    struct ListNode* curr=head;
    while(curr!=NULL && curr->next!=NULL)
    {
        temp=curr->next;
        if(prev!=NULL)
            prev->next=curr->next;
        curr->next=temp->next;
        temp->next=curr;
        if(prev==NULL)
            prev=curr;
        else
            prev=curr;
        curr=curr->next;
    }
    return dummy->next;
}
